test/db/index.js

```diff
@@ -0,0 +1,449 @@
+        await DBHelper.insertEvent(newEventWithExistedTxid);
+        eventCount = await db.Events.count({});
+        eventCount.should.equal(3);
+        const event = await DBHelper.findOneEvent({txid: newEventWithExistedTxid.txid});
+        should.exist(event);
+        expect(event).excluding(["_id", "language"]).to.deep.equal(newEventWithExistedTxid);
+        event.language.should.equal(mockEvent[2].language);
+      })
+    })
+
+    describe('update event', () => {
+      it('update event by txid', async () => {
+        const updateNewEvent = {"txType":"CREATE_EVENT","txid":"0x050497d3ed87c02246ba45ccd965f28e4194aace1167a25e46a2f6e29407d01c","txStatus":"SUCCESS","blockNum":123442,"address":"0x382737120ad2323","ownerAddress":"0xea5854445251393b0e6f0a870799da15c003fc03","version":6,"name":"N","results":["Invalid","12","2"],"numOfResults":3,"centralizedOracle":"0x939592864c0b32d3355b2d15a2203e8343b6d6a","betStartTime":1559689700,"betEndTime":1559692600,"resultSetStartTime":1559692600,"resultSetEndTime":1559864179,"escrowAmount":"110000000","arbitrationLength":86400,"thresholdPercentIncrease":"10","arbitrationRewardPercentage":10,"currentRound":0,"currentResultIndex":255,"consensusThreshold":"10000000000","arbitrationEndTime":0,"status":"OPEN_RESULT_SETTING","language":"en-US"}
+        let event = await DBHelper.findOneEvent({txid: mockEvent[2].txid});
+        expect(event).excluding(["_id"]).to.deep.not.equal(updateNewEvent);
+        await DBHelper.updateEvent(mockEvent[2].txid, updateNewEvent);
+        eventCount = await db.Events.count({});
+        eventCount.should.equal(3);
+        event = await DBHelper.findOneEvent({txid: updateNewEvent.txid});
+        should.exist(event);
+        expect(event).excluding(["_id"]).to.deep.equal(updateNewEvent);
+      })
+
+      it('update event by address', async () => {
+        let event = await DBHelper.findOneEvent({address: '0x382737120ad2323'});
+        expect(event).excluding(["_id"]).to.deep.not.equal(mockEvent[2]);
+        await DBHelper.updateEventByAddress("0x382737120ad2323", mockEvent[2]);
+        eventCount = await db.Events.count({});
+        eventCount.should.equal(3);
+        event = await DBHelper.findOneEvent({txid: mockEvent[2].txid});
+        should.exist(event);
+        expect(event).excluding(["_id"]).to.deep.equal(mockEvent[2]);
+      })
+    })
+
+    describe('event status update', () => {
```

Comment:
please go through the event status tests and add the missing ones. for each of the status change tests you need to test every corner case.

e.g.
updateEventStatusPreBetting(). these are all the tests you should cover:
1. it updates the event to pre-betting status (success case, all fields valid)
2. it does not update events that do not have TX_STATUS.SUCCESS (fail case, where all the other fields are valid except for the txStatus)
3. it does not update events that are already in EVENT_STATUS.PRE_BETTING (fail case, where all other fields are valid except for status)
4. it does not update if currBlockTime = betStartTime (fail case, where all other fields are valid except betStartTime)
4. it does not update events that currentRound > 0 (fail case, where all other fields are valid except currentRound)